llvm/test/Transforms/InstCombine/lshr.ll

```diff
@@ -360,8 +371,187 @@ define <3 x i14> @mul_splat_fold_vec(<3 x i14> %x) {
   ret <3 x i14> %t
 }
 
+define i32 @shl_add_lshr_flag_preservation(i32 %x, i32 %c, i32 %y) {
```

Comment:
Missing test for `add` without nuw flag.

Reply:
Added! @dtcxzyw